### DISCUSSION TITLE
Does Mutation stay in isLoading state till invalidateQuery or refetch query is completed with the return Promise

### DISCUSSION THREAD

**Discussion by nabanita-sarkar** [TanStack/query#6875] [Q&A]
<img width="985" alt="image" src="https://github.com/TanStack/query/assets/36895154/eab3c94d-a6a8-466c-afed-09cf0cb4f31e">

_From @TkDodo 's blog - [Mastering Mutations in React Query](https://tkdodo.eu/blog/mastering-mutations-in-react-query)_

What I am inferring from above is that, the `isLoading` or `isPending` of the mutation will remain true until both the mutation and the invalidated query with query key `['posts', id, 'comments']` is completed. Am I correct to infer that?

Please let me know if this was changed in any recent version or this can't be done when use custom hooks for mutation, because I am unable to recreate this behaviour. (I can share a minimal reproduction if needed)

Thanking in advance. Thanks for this library and all the hard work.

**Comment by TkDodo** [comment]
> Am I correct to infer that?

yes

> I can share a minimal reproduction if needed

please do

**Reply by nabanita-sarkar** [reply]
I tried this in another project. There it does work as expected. Might be some setup issue in the first project.
